chainerx_cc/chainerx/routines/math.cc

```diff
@@ -81,10 +80,25 @@ void BinaryInPlace(Impl&& impl, const Array& x1, Scalar x2) {
     impl(x1, x2, x1);
 }
 
+void CheckArithmeticDtypes(DtypeKind kind1, DtypeKind kind2, DtypeKind out_kind) {
+    if (kind1 == DtypeKind::kBool || kind2 == DtypeKind::kBool || out_kind == DtypeKind::kBool) {
+        throw DtypeError{"Boolean arguments are not supported in arithmetic functions."};
+    }
+    if (out_kind != DtypeKind::kFloat && (kind1 == DtypeKind::kFloat || kind2 == DtypeKind::kFloat)) {
```

Comment:
Why is this check needed?

Reply:
It is because `x1 += x2` (where `x1` is of integer dtype and `x2` is of float dtype) should raise a `DtypeError`.